Core/src/main/java/com/plotsquared/core/generator/HybridGen.java

```diff
@@ -217,34 +222,170 @@ public void generateChunk(@NonNull ScopedQueueCoordinator result, @NonNull PlotA
                     }
                 }
             }
         }
     }
 
+    @Override
+    public boolean populateChunk(final ScopedQueueCoordinator result, final PlotArea settings) {
+        HybridPlotWorld hybridPlotWorld = (HybridPlotWorld) settings;
+        if (!hybridPlotWorld.populationNeeded()) {
+            return false;
+        }
+        // Coords
+        Location min = result.getMin();
+        int bx = min.getX() - hybridPlotWorld.ROAD_OFFSET_X;
+        int bz = min.getZ() - hybridPlotWorld.ROAD_OFFSET_Z;
+        // The relative X-coordinate (within the plot) of the minimum X coordinate
+        // contained in the scoped queue
+        short relativeOffsetX;
+        if (bx < 0) {
+            relativeOffsetX = (short) (hybridPlotWorld.SIZE + (bx % hybridPlotWorld.SIZE));
+        } else {
+            relativeOffsetX = (short) (bx % hybridPlotWorld.SIZE);
+        }
+        // The relative Z-coordinate (within the plot) of the minimum Z coordinate
+        // contained in the scoped queue
+        short relativeOffsetZ;
+        if (bz < 0) {
+            relativeOffsetZ = (short) (hybridPlotWorld.SIZE + (bz % hybridPlotWorld.SIZE));
+        } else {
+            relativeOffsetZ = (short) (bz % hybridPlotWorld.SIZE);
+        }
+        boolean allRoad = true;
+        boolean overlap = false;
+
+        // The X-coordinate of a given X coordinate, relative to the
+        // plot (Counting from the corner with the least positive
+        // coordinates)
+        short[] relativeX = new short[16];
+        boolean[] insideRoadX = new boolean[16];
+        boolean[] insideWallX = new boolean[16];
+        short offsetX = relativeOffsetX;
+        while (offsetX >= hybridPlotWorld.SIZE) {
+            offsetX -= hybridPlotWorld.SIZE;
+        }
+        for (short i = 0; i < 16; i++) {
+            if (offsetX >= hybridPlotWorld.SIZE) {
+                offsetX -= hybridPlotWorld.SIZE;
+                overlap = true;
+            }
+            relativeX[i] = offsetX;
+            if (hybridPlotWorld.ROAD_WIDTH != 0) {
+                boolean insideRoad = offsetX < hybridPlotWorld.PATH_WIDTH_LOWER || offsetX > hybridPlotWorld.PATH_WIDTH_UPPER;
+                boolean insideWall = offsetX == hybridPlotWorld.PATH_WIDTH_LOWER || offsetX == hybridPlotWorld.PATH_WIDTH_UPPER;
+                insideRoadX[i] = insideRoad;
+                insideWallX[i] = insideWall;
+                allRoad &= insideRoad && insideWall;
+            }
+            offsetX++;
+        }
+
+        // The Z-coordinate of a given Z coordinate, relative to the
+        // plot (Counting from the corner with the least positive
+        // coordinates)
+        short[] relativeZ = new short[16];
+        boolean[] insideRoadZ = new boolean[16];
+        boolean[] insideWallZ = new boolean[16];
+        short offsetZ = relativeOffsetZ;
+        while (offsetZ >= hybridPlotWorld.SIZE) {
+            offsetZ -= hybridPlotWorld.SIZE;
```

Comment:
Can't this replaced by an if and a modulo operation?

Reply:
Yeah it should be able to be... I had extracted it out of the for loop below and intended to do just that but forgot